### DISCUSSION TITLE
On Startup: WARNING 'allowipv6' not defined in 'Definition'. Using default one: 'auto'

### DISCUSSION THREAD

**Discussion by jhgarrison** [fail2ban/fail2ban#3699] [General]
This is a newly installed Debian 12.4, and even with an unmodified fail2ban install I get this warning message on startup every time.  This seems to be related to the `allowipv6` option in `fail2ban.conf` being commented out.  Curiously, the commented section containing `allowipv6` is in the `[Default]` section, not `[Definition]`.

I tried adding `allowipv6=yes` to `jail.local` in the `[Default]` section (my system is dual-stack 4/6), but it does not remove the warning.  I then tried adding it in a `[Definition]` section in `jail.local`, also with no effect.

Finally, un-commenting the line in `fail2ban.conf` _does_ prevent the warning, but I'd rather not modify `fail2ban.conf` unless absolutely necessary.  

Is there a way to set the value in `jail.conf` and prevent the warning?

**Comment by sebres** [comment]
This is not an error, just warning...
(However I'd see to disable such warnings, targeting default values, in new version).

> Finally, un-commenting the line in `fail2ban.conf` does prevent the warning, but I'd rather not modify `fail2ban.conf` unless absolutely necessary.

Just add it to local configuration `fail2ban.local` (emphasis on `.local` not `.conf`):
```ini
[DEFAULT]
allowipv6 = auto
```
(don't copy `*.conf` to `*.local`, create empty one and set only parameters you need to overwrite).

> Is there a way to set the value in `jail.conf` ...

No. This parameter works globally and needed directly by start of fail2ban-server (before any jail gets started), so it belongs to fail2ban-config not to jail-config.
But normally one should anyway avoid to change stock `*.conf` (no matter `jail.conf` or `fail2ban.conf`) - there are local configs for that reason.
See https://github.com/fail2ban/fail2ban/wiki/Proper-fail2ban-configuration

**Reply by sebres** [reply]
As for the warning suppression, I forgotten that I already "fixed" it in 432e7e1e93936f09e349e80d94254e5f43d0cc8a (gets out with the next release)

**Reply by codemonkeykong** [reply]
If you have disabled IPv6 in the /etc/sysctl.conf file, you can easily get rid of this warning:
$ sudo nano /etc/fail2ban/fail2ban.conf
--- cut ---
#allowipv6 = auto
allowipv6 = no
--- cut ---
$ sudo systemctl restart fail2ban.service
$ sudo systemctl status fail2ban.service
